RA_Integration/RA_Dlg_GameTitle.h

```diff
@@ -1,24 +1,53 @@
+#ifndef DLG_GAMETITLE_H
+#define DLG_GAMETITLE_H
 #pragma once
 
-#include <wtypes.h>
-#include <map>
-
 #include "RA_Defs.h"
+#include "IRA_Dialog.h"
 
-class Dlg_GameTitle
+class Dlg_GameTitle : public IRA_Dialog
 {
+	using GameTtiles = std::map<std::string, GameID>;
 public:
-	std::string CleanRomName( const std::string& sTryName );
-	static void DoModalDialog( HINSTANCE hInst, HWND hParent, std::string& sMD5InOut, std::string& sEstimatedGameTitleInOut, GameID& nGameIDOut );
+	Dlg_GameTitle();
+	~Dlg_GameTitle() noexcept;
+
+	// This calls DoModal internally
+	void GetInfoFromModal(std::string& sMD5InOut,
+		std::string& sEstimatedGameTitleInOut, GameID nGameIDOut);
+
+	BOOL OnInitDialog(HWND hwnd, HWND hwndFocus, LPARAM lParam) override;
+
 	
-	INT_PTR GameTitleProc( HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam );
+
+
+
+protected:
+	void OnOK(HWND hDlg) override;
+	void OnCommand(HWND hwnd, int id, HWND hwndCtl, UINT codeNotify) override;
+	void OnKnownGames(const UINT codeNotify);
+	void OnGameTitle(HWND hDlg, HWND hwndCtl, UINT codeNotify);
+	// Inherited via IRA_Dialog
+	virtual INT_PTR CALLBACK DialogProc(HWND hDlg, UINT uMsg, WPARAM wParam,
+		LPARAM lParam) override;
 
 private:
-	GameID m_nReturnedGameID;
+	// This strangely got initilized before construction
+	GameID m_nReturnedGameID{ 0 };
 	std::string m_sMD5;
 	std::string m_sEstimatedGameTitle;
 
-	std::map<std::string, GameID> m_aGameTitles;
+	// don't intialize the HWNDs contrustor, weird I know but Win32 isn't in C++
+	// We'd have to make smart pointers for them but that'll take too long
+	HWND hKnownGamesCbo{ nullptr }; 
+	HWND game_title{ nullptr };
+	HWND checksum{ nullptr };
+	GameTtiles m_aGameTitles;
+	bool bUpdatingTextboxTitle{ false };
```

Comment:
please name variables consistently

Reply:
What do you mean by that? I just left variables that were in Hungarian notation alone, the new variables are in the ISO C/C++ convention.

Reply:
Personally, I don't. But I've been taught to maintain whatever style already exists so the entire project is consistent.

Reply:
Sounds good to me